notebooks/problem-sets/quantum_enigma_002.ipynb

```diff
@@ -0,0 +1,418 @@
+    "problem_qc.cx(2,4)\n",
+    "problem_qc.cx(3,4)\n",
+    "problem_qc.barrier(0, 1, 2, 3, 4, 5, 6, 7)\n",
+    "\n",
+    "# Everyone takes note of the answer\n",
+    "problem_qc.cx(4,5)\n",
+    "problem_qc.cx(4,6)\n",
+    "problem_qc.cx(4,7)\n",
+    "problem_qc.barrier(0, 1, 2, 3, 4, 5, 6, 7)\n",
+    "\n",
+    "# Bob checks the parity of the hair color in front of him\n",
+    "problem_qc.cx(2,5)\n",
+    "problem_qc.cx(3,5)\n",
+    "problem_qc.barrier(0, 1, 2, 3, 4, 5, 6, 7)\n",
+    "\n",
+    "# Charlie and Dahlia take note of the answer\n",
+    "problem_qc.cx(5,6)\n",
+    "problem_qc.cx(5,7)\n",
+    "problem_qc.barrier(0, 1, 2, 3, 4, 5, 6, 7)\n",
+    "\n",
+    "# Charlie checks the parity of Dahlia's hair color\n",
+    "problem_qc.cx(3,6)\n",
+    "problem_qc.barrier(0, 1, 2, 3, 4, 5, 6, 7)\n",
+    "\n",
+    "# Dahlia takes note of Charlie's hair color\n",
+    "problem_qc.cx(6,7)\n",
+    "\n",
+    "# Execute the circuit and draw the histogram\n",
+    "\n",
+    "measured_qc = problem_qc.measure_all(inplace=False)\n",
+    "backend = Aer.get_backend('qasm_simulator') # the device to run on\n",
+    "result = backend.run(transpile(measured_qc, backend), shots=1024).result()\n",
+    "counts  = result.get_counts(measured_qc)\n",
+    "plot_histogram(counts)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<!-- ::: q-block.exercise -->\n",
+    "\n",
+    "### Quick quiz\n",
+    "\n",
+    "<!-- ::: q-quiz(goal=\"enigma-002-quiz-5\") -->\n",
+    "\n",
+    "<!-- ::: .question -->\n",
+    "\n",
+    "Can you explain the significance of the first four qubits? (starting from the right)\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: .option(correct) -->\n",
+    "\n",
+    "1. The first 4 qubits each represent the hair color of each player.\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: .option -->\n",
+    "\n",
+    "2. The first 4 qubits each represent the hair color each player is giving as their answer.\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: -->"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<!-- ::: q-block.exercise -->\n",
+    "\n",
+    "### Quick quiz\n",
+    "\n",
+    "<!-- ::: q-quiz(goal=\"enigma-002-quiz-6\") -->\n",
+    "\n",
+    "<!-- ::: .question -->\n",
+    "\n",
+    "Can you explain the significance of the last four qubits? (starting from the right)\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: .option -->\n",
+    "\n",
+    "1. The last 4 qubits each represent the hair color of each player.\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: .option(correct) -->\n",
+    "\n",
+    "2. The last 4 qubits each represent the hair color each player is giving as their answer. \n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: -->\n"
+   ]
+  }
+ ],
```

Comment:
In terms of survey, can we add a survey of just one question that would be "What rating would you give to this problem set?" (on a scale from one to five)? Would it be possible to collect the answers?

If so, I would go this way
-Put the short survey at the end of Enigma 1 at the bottom of the page after Problem 3 (before the Overview of Enigma 2)
-Put the short survey at the end of Enigma 2 at the bottom of the page after Problem 5 (before the Overview of Enigma 3)
-Put the short survey at the end of Enigma 3 at the bottom of the page after Problem 3 (before the Overview of Enigma 4)
Put the long survey after the end of Enigma 4 (as it is now after Enigma 1)

Reply:
Added!